app/components/dashboard/index.tsx

```diff
@@ -40,13 +45,22 @@ export default function Dashboard() {
 const JobItem = ({
   job,
   onRemove,
+  onCancel,
 }: {
   job: PendingJob
   onRemove: (id: JobID) => unknown
+  onCancel: (id: JobID) => unknown
 }) => {
   const progress =
     job.status === 'running' || job.status === 'failed' ? job.progress : 0
+
+  const STUCK_JOB_TIMEOUT_MS = 24 * 60 * 60 * 1000
```

Comment:
If the check is when it was last updated we can probably reduce this significantly right? 3 hours seems reasonable?